### PR TITLE
Alfawise Flash EEPROM, Z Servo Probe

### PR DESCRIPTION
**Pull request by tpruvot**
To do the tests with the flash eeprom changes, which can be set as default... too

### CODE REVIEW

File: Marlin/src/pins/stm32/pins_LONGER3D_LK.h [hunk 1]
```diff
@@ -96,10 +96,14 @@
 //
 // PWM
 //
-#define SERVO0_PIN         PD13  // Open drain PWM pin on the V0G (GND or floating 5V)
-#define SERVO0_PWM_OD            // Comment this if using PE5
+#if HAS_Z_SERVO_PROBE
```
Comment (thinkyhead) [inline Marlin/src/pins/stm32/pins_LONGER3D_LK.h:100]: This is the only pins file to define the servo pins conditionally. Note that Marlin won't implement servos unless `NUM_SERVOS > 0` and customarily we always define these pins (which can be useful for M43 pinsDebug).

Comment (tpruvot) [reply Marlin/src/pins/stm32/pins_LONGER3D_LK.h:100]: im using this pins file as is with a bltouch

File: Marlin/src/pins/stm32/pins_LONGER3D_LK.h [hunk 2]
```diff
@@ -138,21 +142,28 @@
...
-/*
-#define SPI_EEPROM         1   // If commented this will create a file on the SD card as a replacement
-#define SPI_CHAN_EEPROM1   1
-#define SPI_EEPROM1_CS     PC5 // pin 34
-
-//#define EEPROM_SCK  BOARD_SPI1_SCK_PIN   // PA5 pin 30
-//#define EEPROM_MISO BOARD_SPI1_MISO_PIN  // PA6 pin 31
-//#define EEPROM_MOSI BOARD_SPI1_MOSI_PIN  // PA7 pin 32
-*/
+#define E2END (EEPROM_PAGE_SIZE - 1)
```
Comment (tpruvot) [inline Marlin/src/pins/stm32/pins_LONGER3D_LK.h]: may cause issues on SD, using EEPROM_PAGE_SIZE

Comment (thinkyhead) [reply Marlin/src/pins/stm32/pins_LONGER3D_LK.h]: Right. So, I need a default case to set it if it is not defined.

File: Marlin/src/pins/stm32/pins_LONGER3D_LK.h (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/pins/stm32/pins_LONGER3D_LK.h:100]: `BLTOUCH` helpfully sets `NUM_SERVOS` for you if you forget.

File: Marlin/src/pins/stm32/pins_LONGER3D_LK.h (hunk 2 above)
Comment (tpruvot) [reply Marlin/src/pins/stm32/pins_LONGER3D_LK.h]: euh it was :p 0x7FF

File: Marlin/src/pins/stm32/pins_LONGER3D_LK.h [hunk 3]
```diff
@@ -94,12 +94,17 @@
 #define LED_PIN            PC2   // pin 17
 
 //
-// PWM
+// PWM for a servo probe
+// Other servo devices are not supported on this board!
```
Comment (thinkyhead) [inline Marlin/src/pins/stm32/pins_LONGER3D_LK.h:98]: This warning should help users understand the situation.

**Pull request merged**